tools/tests/prepare_sim.py

```diff
@@ -64,9 +64,8 @@ def get_current_runtime_version():
     ios_runtimes = [
         runtime for runtime in runtimes if runtime["platform"] == "iOS"]
 
-    # find the newest runtime. Apple uses semver, so packaging.version allows for stable sorting.
-    ios_runtimes.sort(key=lambda runtime: packaging.version.parse(
-        runtime["version"]), reverse=True)
+    # find the newest runtime
+    ios_runtimes.sort(key=lambda runtime: importlib.metadata.version(runtime["version"]), reverse=True)
```

Comment:
Unrelated to the other changes in this PR. CI is hitting this err in `master` currently (see [example PR](https://github.com/bazel-ios/rules_ios/actions/runs/9387193179/job/25849545106?pr=874)):
```sh
/Users/runner/work/rules_ios/rules_ios/tools/tests/prepare_sim.py:9: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import packaging
Traceback (most recent call last):
  File "/Users/runner/work/rules_ios/rules_ios/tools/tests/prepare_sim.py", line 9, in <module>
    from pkg_resources import packaging
ImportError: cannot import name 'packaging' from 'pkg_resources' (/opt/homebrew/lib/python3.12/site-packages/pkg_resources/__init__.py)
```